src/Symfony/Component/Yaml/Tests/ParserTest.php

```diff
@@ -1751,6 +1751,73 @@ public function testParseMultiLineUnquotedString()
         $this->assertSame(['foo' => 'bar baz foobar foo', 'bar' => 'baz'], $this->parser->parse($yaml));
     }
 
+    /**
+     * @dataProvider getUnquotedMultilineScalarIgnoresCommentsData
+     */
+    public function testUnquotedMultilineScalarIgnoresComments(string $yaml, array $expected)
+    {
+        $this->assertSame($expected, $this->parser->parse($yaml));
+    }
+
+    public static function getUnquotedMultilineScalarIgnoresCommentsData()
+    {
+        yield 'comments interspersed' => [
+            <<<YAML
+                key: unquoted
+                  # this comment should be ignored
+                  next line
+                  # another comment
+                  final line
+                another_key: works
+                YAML,
+            [
+                'key' => 'unquoted next line final line',
+                'another_key' => 'works',
+            ],
+        ];
+
+        yield 'only comments' => [
+            <<<YAML
+                key: unquoted
+                  # this comment should be ignored
+                  # another comment
+                another_key: works
+                YAML,
+            [
+                'key' => 'unquoted',
+                'another_key' => 'works',
+            ],
+        ];
+
+        yield 'blank lines and comments' => [
+            <<<YAML
+                key: unquoted
+                  next line
+
+                  # this comment should be ignored
+                  final line
```

Comment:
I am not sure this is a valid example, all online parser that I found terminate parsing the value as soon as they stumble upon a comment line (see https://yaml-online-parser.appspot.com/?yaml=key%3A+foo%0A+bar%0A+%23+m%0A+foobar%0Akey2%3A+baz&type=json for such an example).

Reply:
see also yaml/yaml#49

Reply:
Hi @xabbuh, thank you for your feedback on the spec compliance ([3.2.3.3](https://yaml.org/spec/1.2.2/#3233-comments)).

I've opened a new PR to address this properly: #62409 which implements what I believe is the correct, spec-compliant behavior.

What do you think of this new approach? Is this the right way to go, or do you think we should prefer to keep a more permissive ([doc-aligned](https://symfony.com/doc/current/reference/formats/yaml.html#comments)) behavior?

> Comments are ignored by the YAML parser and do not need to be indented according to the current level of nesting in a collection.